cmd/main.go

```diff
@@ -49,7 +49,14 @@ const appname = "headers"
 func main() {
 	lf := logger.DefaultLoggerFactory()
 	log := lf.NewLogger("main")
-	c := vconfig.NewViperConfig(appname).
+
+	flagCfg, err := p2pconfig.ParseFlags("")
```

Comment:
What do you mean, by  
> requiring mapping for everything

I thought that viper is handling all the ways we choose for configuration with predefined priority for each of the ways.
So where is this "hell"?

Reply:
If you work with config as a struct you need to map each field with it's value in a file config.
On this stage everything is great, even if you work with nested json file. Just specify mapstructure tag.
Next stage: you want to override some param by env variable. Also ok. You just add a new replacer and you can provide values combining your mapstructure tags with underscore, for example "HTTP_PORT" or "P2P_PROXY" and so on.
But if you want to override something by flags you probably want to have them short, without underscores and so on, you probably want to provide them just like "--port" or "--proxy" or maybe by some short flag like "-P".
And here you need to register all your flags and bind them to viper. And it'll be a lot of dumb code.

Reply:
IMHO, the short flags are only needed for the most commonly used params, the rest can be in form "--p2p.something.something.else" or with `_` instead of `.`
I think that this is not a problem, and for me it seems like a way to go.

Reply:
Or maybe I don't understand it correctly.
I assume that by default each property can be set via flags when using some default pattern to build those flags.
Or you need to register each and every flag?